src/main/java/gov/nih/nci/evs/api/controller/SearchController.java

```diff
@@ -1054,6 +1054,14 @@ public void postInit() throws Exception {
             "Parameter 'pageSize' must be between 1 and 1000 = " + pageSize);
       }
 
+      // If the query has a comment but no newlines, then we have a problem
+      if (query.matches("(?i:.*SELECT.*#.*)") && !query.contains("\n")) {
+        throw new ResponseStatusException(
+            HttpStatus.BAD_REQUEST,
+            "Query appears to contain a comment (#) without newline "
+                + "characters (if using curl use --data-binary instead of -d)");
+      }
+
```

Comment:
I'm assuming this is where we identify the 400 error, correct?

Reply:
yeah, the line before 1061 has a status code for 400 (e.g. BAD_REQUEST)